packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1660,20 +1660,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const savedFlushMode = this.flushMode;
         this.setFlushMode(FlushMode.TurnBased);
 
-        try {
-            this.trackOrderSequentiallyCalls(callback);
-            this.flush();
-            this.setFlushMode(savedFlushMode);
-        } catch(error) {
-            this.closeFn(CreateProcessingError(error, "orderSequentially"));
-            throw error; // throw the original error for the consumer of the runtime
-        }
+        this.trackOrderSequentiallyCalls(callback);
+        this.flush();
+        this.setFlushMode(savedFlushMode);
     }
 
     private trackOrderSequentiallyCalls(callback: () => void): void {
         try {
             this._orderSequentiallyCalls++;
             callback();
+        } catch (error) {
+            if (this._orderSequentiallyCalls === 1) {
+                // In case of nesting, only the top level call should close the container
```

Comment:
I'm not sure I agree with this behavior.
I see two options:
1. exceptions go through callstack to most outer batch. What we do here for inner batches does not matter - container gets closed.
2. Exception is eaten somewhere on the way and does not reach outer batch.

What is the recovery path calling code can make in # 2? I think I do not see any. And if there is none, then we end up with partial batch being committed to model and broadcast to the world.

Reply:
I agree. I will rework this and add a test. If a customer calls orderSequentially recursively and handles/swallows the exception, we will miss the error and not close the container.